classes/order/Order.php

```diff
@@ -1146,7 +1146,7 @@ public static function getByCartId($id_cart)
     {
         $id_order = (int) self::getIdByCartId((int) $id_cart);
 
-        return ($id_order > 0) ? new self($id_order) : null;
+        return ($id_order > 0) ? new static($id_order) : null;
```

Comment:
I'm afraid of possible consequences 🤔

Reply:
Not really, if someone override this controller, it's normal to have the right Order controller loaded, and not the Core's one.